Sources/StreamVideo/Utils/AudioSession/AudioRecorder/CallAudioRecordingStore/CallAudioRecordingStore.swift

```diff
@@ -0,0 +1,195 @@
+        function: StaticString = #function,
+        line: UInt = #line
+    ) {
+        processingQueue.addTaskOperation { [weak self] in
+            guard let self else {
+                return
+            }
+
+            do {
+                try perform(
+                    action,
+                    file: file,
+                    function: function,
+                    line: line
+                )
+            } catch {
+                log.error(
+                    error,
+                    subsystems: .audioRecording,
+                    functionName: function,
+                    fileName: file,
+                    lineNumber: line
+                )
+            }
+        }
+    }
+
+    // MARK: - Private Helpers
+
+    private func perform(
+        _ action: CallAudioRecordingAction,
+        file: StaticString = #file,
+        function: StaticString = #function,
+        line: UInt = #line
+    ) throws {
+        let state = stateSubject.value
+
+        let middleware = middleware
+        let reducers = reducers
+
+        middleware.forEach {
+            $0.apply(
+                state: state,
+                action: action,
+                file: file,
+                function: function,
+                line: line
+            )
+        }
+
+        do {
+            let updatedState = try reducers
+                .reduce(state) {
+                    try $1.reduce(
+                        state: $0,
+                        action: action,
+                        file: file,
+                        function: function,
+                        line: line
+                    )
+                }
+
+            stateSubject.send(updatedState)
+
+            actionLogger.didComplete(
+                action: action,
+                state: updatedState,
+                file: file,
+                function: function,
+                line: line
+            )
+        } catch {
+            actionLogger.didFail(
+                action: action,
+                error: error,
+                file: file,
+                function: function,
+                line: line
+            )
+            throw error
+        }
+    }
+}
+
+extension CallAudioRecordingStore {
+
+    final class ActionLogger {
+
+        private var metersUpdated: [Float] = []
+        private let metersUpdatedLimit: Int = 500
+
+        func didComplete(
+            action: CallAudioRecordingAction,
+            state: State,
+            file: StaticString,
+            function: StaticString,
+            line: UInt
+        ) {
+            if case let .setMeter(value) = action {
+                metersUpdated.append(value)
+                guard
+                    metersUpdated.endIndex == metersUpdatedLimit
+                else {
+                    return
+                }
+
+                let sum = metersUpdated.reduce(0, +)
+                let average = sum / Float(metersUpdatedLimit)
```

Comment:
how frequently do we receive updates? computing average on 500 is translated into how long intervals?

Reply:
As we discussed, this is only being used for logging purposes and to reduce the noise in our logs. Currently it's happening 1-2 seconds